docs/basics/index-lifecycle.md

```diff
@@ -0,0 +1,39 @@
+
+Index Lifecycle
+======================
+
+JanusGraph uses only indexes which have status `ENABLED`. 
+When the index is created it will not be used by JanusGraph until it is enabled. 
+After the index is build you should wait until it is registered (i.e. available) by JanusGraph:
+```java
+//Wait for the index to become available (i.e. wait for status REGISTERED)
+ManagementSystem.awaitGraphIndexStatus(graph, "nameAndAgeIndex").call();
+```
+
+After the index is registered we should either enable the index 
+(if we are sure that the current data should not be indexed by the newly created index) or we should reindex current data 
+so that it would be available in the newly created index.
+```java
+//Reindex the existing data
+mgmt = graph.openManagement();
+mgmt.updateIndex(mgmt.getGraphIndex("nameAndAgeIndex"), SchemaAction.REINDEX).get();
+mgmt.commit();
+```
+
+Index states and transitions
+-----------
+
+![States and transitions](index-lifecycle.png)
+
+## States (SchemaStatus)
+* **INSTALLED** The index is installed in the system but not yet registered with all instances in the cluster
+* **REGISTERED** The index is registered with all instances in the cluster but not (yet) enabled
+* **ENABLED** The index is enabled and in use
+* **DISABLED** The index is disabled and no longer in use
+
+## Actions (SchemaAction)
+* **REGISTER_INDEX** Registers the index with all instances in the graph cluster. After an index is installed, it must be registered with all graph instances
```

Comment:
Same as above, I would add a short sentence like _The following actions can be performed on an index to change its state via `mgmt.updateIndex()`:_